### ISSUE TITLE
训练

### ISSUE THREAD

**Issue by huangmengxiao000** [frederickszk/LRNet#25]
1. Hello, may I ask, are all the settings mentioned in the paper? In preprocessing step, we adopt Dlib [14] to carry out face
and landmark detection (another detector Openface[4] is adopt in the ablation study). In classification procedure, Each RNN in our two-stream network is bidirectional and Consists of GRU (Gated Recurrent Unit) whose number of output units is set to be k = 64. And two fully-connected layers with the number of units to be 64 and 2 are connected to RNN layer's output. A dropout layer with drop rate dr1 = 0.25 is inserted between input and RNN, and another 3 dropout layers with dr2 = 0.5 are used to separate the rest of the layers. These settings are partially based on existing reserach results [26]. In addition, we adopt an 8:2 dataset split, i.e., 80% for training and 20% for testing. Each video is segmented into clips with a fixed length of 60, which sum up to 2 seconds when the fps is 30. We adopt Adam optimizer with lr = 0.001, and batch size is set to be 1024. This classification model will be trained up to 500 epochs.
2. Regarding the question of epoch, the epoch parameters are passed to the model, and in the end is it used in iteration or epoch?Why val_acc fluctuates so much？
3. G1+g2 is to train g1 first and then train g2. After training g1, load the weight of g1 and continue to train g2, or what is the setting of this piece?

**Comment by huangmengxiao000** [comment]
G1+G2 is trained by loading the weight of G1 after training G1

**Comment by huangmengxiao000** [comment]
G1+G2 is trained by loading the weight of G1 after training G1？

**Comment by huangmengxiao000** [comment]
G1+G2 is trained by loading the weight of G1 after training G1？

**Comment by huangmengxiao000** [comment]
"The prediction label of the clip is aggregated for the prediction of the video. How to understand this sentence, what is the way of aggregation? Is it averaging?

**Comment by frederickszk** [comment]
1. 文章版本比较旧了，目前最新更新使用的参数可以参考：`./training/config/` 下的配置文件，已经把所有超参数都统一都记录在这些配置文件中了。关于数据集的划分也可参考`./training/datasets/` 的说明文档，对比论文版本有一定更新。
2. g1和g2是分开训练的，两个是不同网络。具体可以参考`./training/tain.py` 中的133-147行
3. 关于最后结果的聚合方式，是g1和g2分别的结果取平均，获得对每个片段的预测结果。然后再将所有片段的结果取加起来，获得对整个视频的预测概率。具体实现可见：
https://github.com/frederickszk/LRNet/blob/408f28881e2864e8d731143849abf56f961ddefa/training/evaluate.py#L106
以及后面的代码。

**Comment by huangmengxiao000** [comment]
If you choose all, you first train g1 to save the weight, then train g2 and then save the weight, and evaluate these two together when evaluating

**Comment by huangmengxiao000** [comment]
文章中的epoch是指完整训练完一个epoch，也就是一次完整地训练完所有batch吗

**Comment by huangmengxiao000** [comment]
为什么我的val_acc图震荡得厉害

**Comment by huangmengxiao000** [comment]
代码中g1+g2的效果就是论文中g1+g2要的结果吗

**Comment by huangmengxiao000** [comment]
关于最后结果的聚合方式，是g1和g2分别的结果取平均，获得对每个片段的预测结果。然后再将所有片段的结果取加起来，获得对整个视频的预测概率。所有片段的结果取加起来求平均吗？

**Comment by huangmengxiao000** [comment]
不好意思打扰了，您能给我说一些下面这些的含义吗
feature_size: 136
lm_dropout_rate: 0.1
rnn_unit: 32
num_layers: 1
rnn_dropout_rate: 0
fc_dropout_rate: 0.5
res_hidden: 64

**Comment by huangmengxiao000** [comment]
不好意思打扰了，现在使用的model参数是下面这些吗？
feature_size: 136
lm_dropout_rate: 0.1
rnn_unit: 32
num_layers: 1
rnn_dropout_rate: 0
fc_dropout_rate: 0.5
res_hidden: 64
classify里面的模型是tensorflow用的？
model = K.Sequential([
        layers.InputLayer(input_shape=(block_size, 136)),
        layers.Dropout(0.25),
        layers.Bidirectional(layers.GRU(RNN_UNIT)),
        layers.Dropout(DROPOUT_RATE),
        layers.Dense(64, activation='relu'),
        layers.Dropout(DROPOUT_RATE),
        layers.Dense(2, activation='softmax')
    ])

**Comment by huangmengxiao000** [comment]
论文中提到的k=64是在哪里设置的，In classification procedure,
Each RNN in our two-stream network is bidirectional and
consists of GRU (Gated Recurrent Unit) whose number of
output units is set to be k = 64.
在pytorch中用的模型参数是下面这些吗
feature_size: 136
lm_dropout_rate: 0.1
rnn_unit: 32
num_layers: 1
rnn_dropout_rate: 0
fc_dropout_rate: 0.5
res_hidden: 64

**Comment by huangmengxiao000** [comment]
论文中提到的k=64是在哪里设置的，In classification procedure,
Each RNN in our two-stream network is bidirectional and
consists of GRU (Gated Recurrent Unit) whose number of
output units is set to be k = 64.
在pytorch中用的模型参数是下面这些吗，这些参数的含义是什么
feature_size: 136
lm_dropout_rate: 0.1
rnn_unit: 32
num_layers: 1
rnn_dropout_rate: 0
fc_dropout_rate: 0.5
res_hidden: 64

**Comment by langhua11** [comment]
The dataset directory: [./datasets\Origin\c23](https://file+.vscode-resource.vscode-cdn.net/d%3A/project_xm/LRNet-main/training/datasets/Origin/c23) does not exist. Please check and restart. 
 这个怎么解决？

**Comment by frederickszk** [comment]
> The dataset directory: [./datasets\Origin\c23](https://file+.vscode-resource.vscode-cdn.net/d%3A/project_xm/LRNet-main/training/datasets/Origin/c23) does not exist. Please check and restart. 这个怎么解决？

@langhua11 
需要先下载数据集，然后放在正确的路径下。详情可以参考[这个页面](https://github.com/frederickszk/LRNet/tree/main/training/datasets)

**Comment by langhua11** [comment]
> > The dataset directory: [./datasets\Origin\c23](https://file+.vscode-resource.vscode-cdn.net/d%3A/project_xm/LRNet-main/training/datasets/Origin/c23) does not exist. Please check and restart. 这个怎么解决？
> 
> @langhua11 需要先下载数据集，然后放在正确的路径下。详情可以参考[这个页面](https://github.com/frederickszk/LRNet/tree/main/training/datasets)

感谢你的回复，你的数据集里没有c23文件好像

**Comment by frederickszk** [comment]
> > > The dataset directory: [./datasets\Origin\c23](https://file+.vscode-resource.vscode-cdn.net/d%3A/project_xm/LRNet-main/training/datasets/Origin/c23) does not exist. Please check and restart. 这个怎么解决？
> > 
> > 
> > @langhua11 需要先下载数据集，然后放在正确的路径下。详情可以参考[这个页面](https://github.com/frederickszk/LRNet/tree/main/training/datasets)
> 
> 感谢你的回复，你的数据集里没有c23文件好像

@langhua11 c23文件夹是各个不同数据集下的子目录（它的上层文件可能是`Origin`, `DF`, `FS`等），从网盘下载了完整的数据集解压后应该能看到
